### PR TITLE
pkg/db(ticdc): add pebble db

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve? 

Add pebble as a db backend used in leveldb sorter.

### Check List 

Tests 

 - Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: pkg/db/db_test.go [hunk 1]
```diff
@@ -30,9 +31,17 @@ func TestDB(t *testing.T) {
 	cfg := config.GetDefaultServerConfig().Clone().Debug.DB
 	cfg.Count = 1
 
-	ldb, err := leveldb.Open(storage.NewMemStorage(), &opt.Options{})
+	db, err := OpenDB(ctx, 1, filepath.Join(t.TempDir(), "1"), cfg)
```
Comment (hi-rustin) [inline pkg/db/db_test.go:34]: This may need to be renamed to OpenLevelDB?

File: pkg/db/pebble.go [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+	*pebble.Batch
+}
+
+var _ Batch = (*pebbleBatch)(nil)
+
+func (b pebbleBatch) Put(key, value []byte) {
+	_ = b.Batch.Set(key, value, pebble.NoSync)
+}
+
+func (b pebbleBatch) Delete(key []byte) {
```
Comment (hi-rustin) [inline pkg/db/pebble.go:193]: It's a bit strange that lint should be complaining that you didn't write a comment for a public method, right?

**Review by hi-rustin — COMMENTED** [review]
rest LGTM

File: pkg/db/pebble.go (hunk 2 above)
Comment (overvenus) [reply pkg/db/pebble.go:193]: Yeah, it may due to the private receiver `pebbleBatch`.

**Review by hi-rustin — APPROVED** [review]
LGTM

File: pkg/db/pebble.go [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+	"go.uber.org/zap"
+)
+
+type pebbleLogger struct{ id int }
+
+var _ pebble.Logger = (*pebbleLogger)(nil)
+
+func (logger *pebbleLogger) Infof(format string, args ...interface{}) {
+	// Do not output low-leveldb pebble log to TiCDC log.
+	log.Debug(fmt.Sprintf(format, args...), zap.Int("db", logger.id))
```
Comment (amyangfei) [inline pkg/db/pebble.go:41]: `fmt.Sprintf` will be always called here, better to make it lazily.

**Review by amyangfei — APPROVED** [review]

File: pkg/db/pebble.go [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+	"github.com/pingcap/ticdc/pkg/retry"
+	"go.uber.org/zap"
+)
+
+type pebbleLogger struct{ id int }
+
+var _ pebble.Logger = (*pebbleLogger)(nil)
+
+func (logger *pebbleLogger) Infof(format string, args ...interface{}) {
+	// Do not output low-leveldb pebble log to TiCDC log.
```
Comment (overvenus) [inline pkg/db/pebble.go:40]:
```suggestion
	// Do not output low-level pebble log to TiCDC log.
```

**Comment by overvenus** [comment]
Integration test fails due to https://github.com/tikv/tikv/issues/11543

**Pull request merged**